input/input.go

```diff
@@ -0,0 +1,252 @@
+	ErrMissingRepositoryType  = errors.New("missing repository type")
+	ErrMissingTargetNameList  = errors.New("missing target name in audit job")
+	ErrMissingTargetName      = errors.New("missing target name")
+	ErrMissingTargetType      = errors.New("missing target type")
+	ErrPollingIntervalFormat  = errors.New("polling interval format is incorrect")
+)
+
+var (
+	log logger.Logger
+)
+
+func init() {
+	log = logger.GetLogger()
+}
+
+// Config represents the data parsed from config.yaml.
+type Config struct {
+	// Loglevel defines the logging level for the log file. <OPTIONAL>
+	Loglevel string `yaml:"loglevel,omitempty" json:"loglevel,omitempty"`
+	// Logpath represents the path where log file will be generated. <OPTIONAL>
+	Logpath string `yaml:"logpath,omitempty" json:"logpath,omitempty"`
+	// Auditjobs is list of auditjobs. <REQUIRED>
+	AuditJobs []AuditJob `yaml:"auditJobs" json:"auditJobs"`
+	// Targets is list of targets. <REQUIRED>
+	Targets []Target `yaml:"targets" json:"targets"`
+}
+
+// validate checks for mandatory fields in config struct and returns error in case of something is missing.
+func (c *Config) validate() error {
+	// checking if at least one job is present.
+	if len(c.AuditJobs) == 0 {
+		return ErrMissingAuditJob
+	}
+	// checking if at least one target is present.
+	if len(c.Targets) == 0 {
+		return ErrMissingTarget
+	}
+	for _, j := range c.AuditJobs {
+		// checking if audit job name is not empty.
+		if j.Name == "" {
+			return ErrMissingAuditJobName
+		}
+		// checking if access token is not empty.
+		if j.AccessToken == "" {
+			return ErrMissingAccessToken
+		}
+		// checking if either username or email is not empty.
+		if j.Username == "" && j.Email == "" {
+			return ErrMissingUsernameEmail
+		}
+		// checking if repository name is not empty.
+		if j.RepositoryName == "" {
+			return ErrMissingRepositoryName
+		}
+		// checking if repository url is not empty.
+		if j.RepositoryURL == "" {
+			return ErrMissingRepositoryURL
+		}
+		// checking if repository type is not empty.
+		if j.RepositoryType == "" {
+			return ErrMissingRepositoryType
+		}
+		// checking if any target is defined in output.
+		if len(j.TargetName) == 0 {
+			return ErrMissingTargetName
+		}
+		// checking if polling interval is not empty.
+		if j.PollingInterval == "" {
+			return ErrMissingPollingInterval
+		}
+		if j.PollingInterval != "" {
+			return checkPollingIntervalFormat(j.PollingInterval)
+		}
+	}
+	for _, t := range c.Targets {
+		// checking if target name is missing.
+		if t.Name == "" {
+			return ErrMissingTargetName
+		}
+		// checking if target type is missing.
+		if t.Type == "" {
+			return ErrMissingTargetType
+		}
+	}
+	return nil
+}
+
+// populateDefaultValues puts default values to optional fields in config.
+func (c *Config) populateDefaultValues() {
+	for i := range c.AuditJobs {
+		if len(c.AuditJobs[i].Branches) == 0 {
+			c.AuditJobs[i].Branches = append(c.AuditJobs[i].Branches, DefaultBranch)
```

Comment:
Add debug logs, like branch is not configured, setting default branch

Reply:
done